File: src/main/java/cs/android/view/list/CSRowView.java
package cs.android.view.list;

import android.view.View;

import cs.android.viewbase.CSLayoutId;
import cs.android.viewbase.CSView;

/**
 * Created by renetik on 08/12/17.
 */

public class CSRowView<T> extends CSView<View> implements CSIRowView<T> {
    private T _row;

    public CSRowView(CSListController parent, CSLayoutId layout) {
        super(parent.asAdapterView(), layout);
    }

    public T row() {
        return _row;
    }

    public void row(T row) {
        _row = row;
        onLoad(_row);
    }

    protected void onLoad(T row) {
    }
}

File: library/renetik-controller-listview/src/main/java/CSRowView.kt
package renetik.android.listview

import android.view.View
import renetik.android.base.CSLayoutId
import renetik.android.base.CSView
import renetik.android.controller.base.CSViewController

open class CSRowView<RowType : Any> : CSView<View> {

    var onLoad: ((CSRowView<RowType>).(RowType) -> Unit)? = null

    constructor(parent: CSListController<RowType, *>, layout: CSLayoutId,
                onLoad: ((CSRowView<RowType>).(RowType) -> Unit)? = null)
            : super(parent, layout) {
        this.onLoad = onLoad
    }

    constructor(parent: CSViewController<*>, onLoad: ((CSRowView<RowType>).(RowType) -> Unit)? = null)
            : super(parent) {
        this.onLoad = onLoad
    }

    lateinit var data: RowType
    var index = -1

    fun load(index: Int, data: RowType) {
        this.index = index
        this.data = data
        onLoad(data)
    }

    protected open fun onLoad(row: RowType) {
        onLoad?.invoke(this, row)
    }
}
